src/authorization/permissions/WrappedMethodModesExtractor.ts

```diff
@@ -0,0 +1,32 @@
+import type { Operation } from '../../http/Operation';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import { ModesExtractor } from './ModesExtractor';
+import type { AccessMode } from './Permissions';
+
+/**
+ * Only allows {@link Operation}s with the specified methods.
+ * If the method matches the Operation will be sent to the provided extractor.
+ */
+export class WrappedMethodModesExtractor extends ModesExtractor {
```

Comment:
So at the very least, I would turn this into `ConditionalOperationHandler`, since there seems to be no need to use the specific `ModesExtractor` subclass.

I wonder if we can have a pattern similar to `IfNeededConverter`, which can be used both in a wrapped and a chained way (note the default value for the constructor argument).

Reply:
Can definitely make this generic (will also need this for the N3 Patch PR anyway). Personally I'm more of a fan of the wrapped solution instead of the fallthrough solution. I think that was also my opinion for the `IfNeededConverter` 😄